alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php

```diff
@@ -392,11 +393,15 @@ private function getAutoEmbedCode($targetId = null)
 		}
 
 		$entry_id = $this->getRequestParameter(self::ENTRY_ID_PARAM_NAME);
-		if (!$entry_id)
-		{
-			KExternalErrors::dieError(KExternalErrors::MISSING_PARAMETER, "Entry ID not defined");
+        $playlist_id = $this->getRequestParameter(self::PLAYLIST_ID_PARAM_NAME);
+        $loadContentMethod = null;
+		if (!is_null($entry_id)) {
+		    $loadContentMethod = "loadMedia({" . "entryId: \"" . $entry_id . "\"" . "});";
+		} elseif (!is_null($playlist_id)) {
+		    $loadContentMethod = "loadPlaylist({" . "playlistId: \"" . $playlist_id . "\"" . "});";
```

Comment:
why add the extra concatenations between strings?
Can be reduced for clarity to:
```php
"loadMedia({entryId: \"" . $entry_id . "\"});";
```
and
```php
"loadPlaylist({playlistId: \"" . $playlist_id . "\"});";
```

Also, as we are building javascript strings in php I would like to avoid as much as possible the javascript being assembled in separate parts of the file.
It is hard enough to manage writing without syntax errors at current state.